components/calendar/src/any_calendar.rs

```diff
@@ -560,14 +560,11 @@ impl Calendar for AnyCalendar {
 impl AnyCalendar {
     /// Constructs an AnyCalendar for a given calendar kind from compiled data.
     ///
-    /// As this requires a valid [`AnyCalendarKind`] to work, it does not do any kind of locale-based
-    /// fallbacking. If this is desired, use [`Self::try_new()`].
-    ///
     /// ✨ *Enabled with the `compiled_data` Cargo feature.*
     ///
     /// [📚 Help choosing a constructor](icu_provider::constructors)
     #[cfg(feature = "compiled_data")]
-    pub const fn new_for_kind(kind: AnyCalendarKind) -> Self {
+    pub const fn new(kind: AnyCalendarKind) -> Self {
```

Comment:
Nit: Keep the method named `new_for_kind` and we just don't have a `new`. I still stand by my previous position that `new` and `try_new` should take Preferences by convention.

Reply:
I'm okay with that, though I think just `new` is fine here.

Reply:
eh, it is nice that `try_new_from_buffer_provider` doesn't need to be `try_new_for_kind_with_buffer_provider`

Let's keep it as-is